src/hazelcore/Builtins.re

```diff
@@ -1,6 +1,46 @@
+module Impls = {
```

Comment:
Let's create a single mapping from the builtin name to both its type and its evaluation behavior, so that there is one localized place where each builtin is implemented.